src/components/menu/editor-toolbar-menu.jsx

```diff
@@ -9,15 +11,24 @@ import CellMenuSubsection from './cell-menu-subsection'
 import SavedNotebooksAndExamplesSubsection from './saved-notebooks-and-examples-subsection'
 import ViewModeToggleSubsection from './view-mode-toggle-subsection'
 
-export default class EditorToolbarMenu extends React.Component {
+export class EditorToolbarMenuUnconnected extends React.Component {
+  static propTypes = {
+    isAuthenticated: PropTypes.bool.isRequired,
+  }
+
   render() {
+    let elem = <NotebookMenuItem task={tasks.loginGithub} />
+    if (this.props.isAuthenticated) {
+      elem = <NotebookMenuItem task={tasks.exportGist} />
+    }
     return (
       <NotebookIconMenu>
         <NotebookMenuItem task={tasks.createNewNotebook} />
         <NotebookMenuItem task={tasks.saveNotebook} />
         <NotebookMenuItem task={tasks.exportNotebook} />
         <NotebookMenuItem task={tasks.exportNotebookAsReport} />
         <NotebookMenuItem task={tasks.clearVariables} />
+        {elem}
```

Comment:
Could we express this a bit more concisely? Not sure if this will work but maybe:

```jsx
{
   this.props.isAuthenticated ? (<NotebookMenuItem task={tasks.loginGithub} />) : (<NotebookMenuItem task={tasks.exportGist} />)
}
```